Use structure that matches the used stat function variant on Windows.

diff -urN netcdf-c-4.9.3/include/ncpathmgr.h.orig netcdf-c-4.9.3/include/ncpathmgr.h
--- netcdf-c-4.9.3/include/ncpathmgr.h.orig	2025-02-07 22:40:00.000000000 +0100
+++ netcdf-c-4.9.3/include/ncpathmgr.h	2025-06-10 22:06:51.111348788 +0200
@@ -190,7 +190,7 @@
 EXTERNL int NCmkstemp(char* buf);
 
 #ifdef HAVE_SYS_STAT_H
-EXTERNL int NCstat(const char* path, struct stat* buf);
+EXTERNL int NCstat(const char* path, struct __stat64* buf);
 #endif
 #ifdef HAVE_DIRENT_H
 EXTERNL DIR* NCopendir(const char* path);
diff -urN netcdf-c-4.9.3/libdispatch/dpathmgr.c.orig netcdf-c-4.9.3/libdispatch/dpathmgr.c
--- netcdf-c-4.9.3/libdispatch/dpathmgr.c.orig	2025-06-10 22:03:16.134410883 +0200
+++ netcdf-c-4.9.3/libdispatch/dpathmgr.c	2025-06-10 22:03:27.072496140 +0200
@@ -686,7 +686,7 @@
 #ifdef HAVE_SYS_STAT_H
 EXTERNL
 int
-NCstat(const char* path, struct stat* buf)
+NCstat(const char* path, struct __stat64* buf)
 {
     int status = 0;
     char* cvtpath = NULL;
diff -urN netcdf-c-4.9.3/libnczarr/zmap_file.c.orig netcdf-c-4.9.3/libnczarr/zmap_file.c
--- netcdf-c-4.9.3/libnczarr/zmap_file.c.orig	2025-02-07 22:40:00.000000000 +0100
+++ netcdf-c-4.9.3/libnczarr/zmap_file.c	2025-06-10 22:20:03.858502334 +0200
@@ -685,7 +685,11 @@
 platformtestcontentbearing(const char* canonpath)
 {
     int ret = 0;
+#if defined(_WIN32)
+    struct __stat64 buf;
+#else
     struct stat buf;
+#endif
     
     ZTRACE(6,"map=%s canonpath=%s",zfmap->map.url,canonpath);
 
@@ -1056,7 +1060,7 @@
 {
     int ret = NC_NOERR;
     off_t size, newsize;
-    struct stat statbuf;    
+    struct stat statbuf;
     
     assert(fd && fd->fd >= 0);
     
@@ -1166,7 +1170,11 @@
 verify(const char* path, int isdir)
 {
     int ret = 0;
+#if defined(_WIN32)
+    struct __stat64 buf;
+#else
     struct stat buf;
+#endif
 
     ret = NCaccess(path,ACCESS_MODE_EXISTS);
     if(ret < 0)
@@ -1182,7 +1190,11 @@
 verifykey(const char* key, int isdir)
 {
     int ret = 0;
+#if defined(_WIN32)
+    struct __stat64 buf;
+#else
     struct stat buf;
+#endif
 
     if(key[0] == '/') key++; /* Want relative name */
 
diff -urN netcdf-c-4.9.3/libnczarr/zplugins.c.orig netcdf-c-4.9.3/libnczarr/zplugins.c
--- netcdf-c-4.9.3/libnczarr/zplugins.c.orig	2025-02-07 22:40:00.000000000 +0100
+++ netcdf-c-4.9.3/libnczarr/zplugins.c	2025-06-10 22:22:44.198048063 +0200
@@ -221,7 +221,11 @@
     int ret = NC_NOERR;
     size_t i,j;
     struct NCglobalstate* gs = NC_getglobalstate();
+#if defined(_WIN32)
+    struct __stat64 buf;
+#else
     struct stat buf;
+#endif
     NClist* dirs = nclistnew();
     char* defaultpluginpath = NULL;
 

